third_party/move/move-bytecode-verifier/src/signature_v2.rs

```diff
@@ -812,7 +812,13 @@ impl<'a, const N: usize> SignatureChecker<'a, N> {
         for (offset, instr) in code.code.iter().enumerate() {
             let map_err = |res: PartialVMResult<()>| {
                 res.map_err(|err| {
-                    err.append_message_with_separator(' ', format!("at offset {}", offset))
+                    err.append_message_with_separator(
+                        ' ',
+                        format!(
+                            "missing abilities for `{:?}` at code offset {}",
```

Comment:
Yes it is. The signature checker does verify ability constraints here.

Reply:
Why doesn't this message show up in any tests?

Reply:
Because we do not have any such tests which generate 'wrong'  bytecode.

But you an see (in the first commit, on Github not here) [how the error looks like](https://github.com/aptos-labs/aptos-core/pull/14987/commits/5bb29c3192761757d75ab05b93bcb2922ab5ebf0#diff-79d527a70ea62fc1effb20aaa589375cccb0c36c321b50db1979ea88d8d5e5c4) before this bug was fixed, but with the error message.

The reason BTW why I added this error message is that the bug was immediately associated by users with enums, but actually had nothing to do with them. By printing some more info, we can avoid misattribution.